pkg/file/pipeline/bmt/bmt.go

```diff
@@ -39,24 +34,22 @@ func (w *bmtWriter) ChainWrite(p *pipeline.PipeWriteArgs) error {
 	if len(p.Data) < swarm.SpanSize {
 		return errInvalidData
 	}
-	w.b.Reset()
-	err := w.b.SetSpanBytes(p.Data[:swarm.SpanSize])
+	hasher := bmtpool.Get()
+	err := hasher.SetSpanBytes(p.Data[:swarm.SpanSize])
 	if err != nil {
 		return err
 	}
-	_, err = w.b.Write(p.Data[swarm.SpanSize:])
+	_, err = hasher.Write(p.Data[swarm.SpanSize:])
 	if err != nil {
 		return err
 	}
-	p.Ref = w.b.Sum(nil)
+	p.Ref = hasher.Sum(nil)
+	bmtpool.Put(hasher)
```

Comment:
This "hasher" will never be returned to the pool if some error occurs (however unlikely these two operation will fail).